### PR TITLE
fix: use `react-jsx` transform

### PR DESCRIPTION
**Pull request by Lukas742**

### CODE REVIEW

File: packages/charts/src/components/TimelineChart/TimelineChart.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import type { CommonProps } from '@ui5/webcomponents-react';
 import { throttle } from '@ui5/webcomponents-react-base';
-import type { CSSProperties, ReactNode } from 'react';
-import React, { forwardRef, useEffect, useRef, useState } from 'react';
+import { forwardRef, useEffect, useRef, useState } from 'react';
+import type * as React from 'react';
```
Comment (MarcusNotheis) [inline packages/charts/src/components/TimelineChart/TimelineChart.tsx:4]: I would prefer to stay in our common pattern here: `import type { CSSProperties, ReactNode } from 'react';`

File: packages/charts/src/components/TimelineChart/chartbody/TimelineChartRow.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { throttle } from '@ui5/webcomponents-react-base';
-import type { CSSProperties } from 'react';
-import React, { useEffect, useRef, useState } from 'react';
+import { useEffect, useRef, useState } from 'react';
+import type * as React from 'react';
```
Comment (MarcusNotheis) [inline packages/charts/src/components/TimelineChart/chartbody/TimelineChartRow.tsx:3]: Pattern: `import type { CSSProperties } from 'react';`

**Review by MarcusNotheis — CHANGES_REQUESTED** [review]
Two minor adjustments needed to stay consistent with our common `import type` pattern.
Other than that: I would prefer to have to commit marked as at least `fix` to make sure it will be available in the release notes.

File: packages/charts/src/components/TimelineChart/TimelineChart.tsx (hunk 1 above)
Comment (Lukas742) [reply packages/charts/src/components/TimelineChart/TimelineChart.tsx:4]: This is not possible here, as the component imports `MouthEvent` both from typescript and React. 

```ts
React.MouseEvent<HTMLDivElement, MouseEvent>
```

Using the named import of `CSSProperties and `ReactNode` and keeping the namespace import for `MouseEvent` also doesn't work as we would then have multiple `type` imports of the same package.

**Review by MarcusNotheis — APPROVED** [review]

**Pull request merged**